agent/containermetadata/manager.go

```diff
@@ -117,8 +118,10 @@ func (manager *metadataManager) Create(config *docker.Config, hostConfig *docker
 
 // Update updates the metadata file after container starts and dynamic metadata is available
 func (manager *metadataManager) Update(dockerID string, task *api.Task, containerName string) error {
+	ctx, cancel := context.WithTimeout(context.TODO(), inspectContainerTimeout)
```

Comment:
You should modify `Update(` to expect an arg for `context.Context` as well. The `DockerTaskEngine` struct already has a context, which it can pass in here.

Reply:
OK, will change accordingly.